src/pages/Bridge/IssueForm/index.tsx

```diff
@@ -132,6 +141,8 @@ const IssueForm = (): JSX.Element | null => {
   const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
   const [submitError, setSubmitError] = React.useState<Error | null>(null);
   const [submittedRequest, setSubmittedRequest] = React.useState<Issue>();
+  const [selectVaultManually, setSelectVaultManually] = React.useState<boolean>(false);
```

Comment:
As long as it works, not a blocker but actually react-hook-form is mostly based on uncontrolled component by passing ref and here it's used as controlled component. So error handling is manually controlled in React.useEffect. Please have a look at how it's used in `TokenField`: validation, data binding, state management and error handling are organized/controlled via react-hook-form mechanism without introducing other things like React.useEffect, React.useState, etc.

In consideration of timeline and shipping new features quickly, I do not have a strong opinion. It's better to ask @tomjeatt's opinion whether to update or not.

Reply:
@crypto-engineer @peterslany I'm happy with this for now - this feature is going to be superseded by Jay's updated vault selection flow very soon and at that point we'll integrate properly with react-hook-forms.

Reply:
Yes, the Select component is controlled, I discussed it before with Tom and he approved this approach. It would require more effort to pass ref into the Select component since it's not using a native `<select>` HTML element I think.

Reply:
As long as we have this in mind, I'm fine.
Yes, properly integrating with react-hook-form could take some research and investigation by reading the docs but I'd say it's worth doing that because we can avoid handling manually and make the modules compact and implemented in a declarative way which should result in less code and less bugs.

I'm fine with the current approach.